src/napari_phasors/lifetime_tab.py

```diff
@@ -0,0 +1,552 @@
+        self.colormap_contrast_limits = self.lifetime_layer.contrast_limits
+        self.lifetime_layer.events.contrast_limits.connect(
+            self._on_colormap_changed
+        )
+
+    def _update_lifetime_histogram(self):
+        """Update the histogram plot with the current histogram values."""
+        self.hist_ax.clear()
+        self.hist_ax.plot(self.bin_centers, self.counts, color='none', alpha=0)
+
+        # Check if colormap is available, if not use a default one
+        if (
+            self.lifetime_colormap is None
+            or self.colormap_contrast_limits is None
+        ):
+            # Use a default colormap if napari colormap is not available yet
+            cmap = plt.cm.turbo
+            norm = plt.Normalize(
+                vmin=(
+                    np.min(self.bin_centers)
+                    if len(self.bin_centers) > 0
+                    else 0
+                ),
+                vmax=(
+                    np.max(self.bin_centers)
+                    if len(self.bin_centers) > 0
+                    else 1
+                ),
+            )
+        else:
+            # Create the colormap by linear combination of the napari cmap
+            cmap = LinearSegmentedColormap.from_list(
+                'custom_cmap', self.lifetime_colormap
+            )
+            norm = plt.Normalize(
+                vmin=self.colormap_contrast_limits[0],
+                vmax=self.colormap_contrast_limits[1],
+            )
+
+        for count, bin_start, bin_end in zip(
+            self.counts, self.bin_edges[:-1], self.bin_edges[1:]
+        ):
+            bin_center = (bin_start + bin_end) / 2
+            color = cmap(norm(bin_center))
+            self.hist_ax.fill_between(
+                [bin_start, bin_end], 0, count, color=color, alpha=0.7
+            )
+
+        # Apply styling after clearing/plotting
+        self.style_histogram_axes()
+
+        # Refresh the canvas to show the updated histogram
+        self.hist_fig.canvas.draw_idle()
+
+    def _on_harmonic_changed(self):
+        """Callback whenever the harmonic selector changes."""
+        self.plot_lifetime_histogram()
+
+    def _on_colormap_changed(self, event):
+        """Callback whenever the colormap changes."""
+        if getattr(self, '_updating_contrast_limits', False):
+            return
+
+        layer = event.source
+        self.lifetime_colormap = layer.colormap.colors
+        self.colormap_contrast_limits = layer.contrast_limits
+
+        # Only update the histogram visualization, don't change slider or data
+        self._update_lifetime_histogram()
+
+    def _on_image_layer_changed(self):
+        """Callback whenever the image layer with phasor features changes."""
+        layer_name = (
+            self.parent_widget.image_layer_with_phasor_features_combobox.currentText()
+        )
+        if layer_name == "":
+            self.histogram_widget.hide()
+            return
+
```

Comment:
This could also be removed and always reset the lifetime widget if the Image Layer is changed, something like the line below.
```suggestion
        # Hide histogram and clears previous data
        self.lifetime_type_combobox.setCurrentText("None")
```